core/src/scopyhomeaddpage.cpp

```diff
@@ -1,68 +1,306 @@
+		if (uri.isEmpty()) {
+			ui->btnVerify->setEnabled(false);
+		} else {
+			ui->btnVerify->setEnabled(true);
+		}
+	});
+
+	connect(this, &ScopyHomeAddPage::contextDataAvailable, this, [=](QMap<QString, QString> ctxInfo){
+		for (QString key : ctxInfo.keys()) {
+			deviceInfoPage->update(key, ctxInfo[key]);
+		}
+	},Qt::QueuedConnection);
+}
+
+ScopyHomeAddPage::~ScopyHomeAddPage()
+{
+	delete ui;
+	if (deviceImpl) {
+		delete deviceImpl;
+	}
+	if (libSerialSupport) {
+		delete libSerialSupport;
+	}
+
+}
+
+void ScopyHomeAddPage::initAddPage()
+{
+	bool hasLibSerialPort = libSerialSupport->hasLibSerialSupport();
+	bool serialBackEnd = iio_has_backend("serial");
+
+	QRegExp re("[5-9]{1}(n|o|e|m|s){1}[1-2]{1}(x|r|d){1}$");
+	QRegExpValidator *validator = new QRegExpValidator(re, this);
+
+	ui->editSerialFrameConfig->setValidator(validator);
+	ui->serialSettingsWidget->setEnabled(hasLibSerialPort && serialBackEnd);
+	ui->btnAdd->setProperty("blue_button", QVariant(true));
+	ui->btnVerify->setProperty("blue_button", QVariant(true));
+	ui->btnScan->setProperty("blue_button", QVariant(true));
+	ui->btnBack->setProperty("blue_button", QVariant(true));
+	ui->btnVerify->setEnabled(false);
+	ui->stackedWidget->setCurrentWidget(ui->deviceDetectionPage);
+
+	for (int baudRate : availableBaudRates) {
+		ui->comboBoxBaudRate->addItem(QString::number(baudRate));
+	}
+}
+
+void ScopyHomeAddPage::initSubSections()
+{
+	ui->subSeparatorPlugins->setLabel("Compatible plugins");
+	ui->subSeparatorPlugins->getContentWidget()->layout()->setSpacing(10);
+
+	ui->subSeparatorDevInfo->setLabel("Device info");
+	deviceInfoPage = new InfoPage(ui->subSeparatorDevInfo->getContentWidget());
+	deviceInfoPage->setAdvancedMode(false);
+	deviceInfoPage->setSizePolicy(QSizePolicy::Expanding,QSizePolicy::Expanding);
+	ui->subSeparatorDevInfo->getContentWidget()->layout()->addWidget(deviceInfoPage);
+	QSpacerItem *vSpacer = new QSpacerItem(40, 20, QSizePolicy::Minimum, QSizePolicy::Expanding);
+	ui->subSeparatorDevInfo->getContentWidget()->layout()->addItem(vSpacer);
+}
+
+void ScopyHomeAddPage::findAvailableSerialPorts()
+{
+	if (scanParamsList.contains("serial:")) {
+		QVector<QString> portsName = libSerialSupport->getSerialPortsName();
+		ui->comboBoxSerialPort->clear();
+		if (!portsName.empty()) {
+			for (QString port : portsName) {
+				ui->comboBoxSerialPort->addItem(port);
+			}
+		}
+	}
 }
 
+void ScopyHomeAddPage::verifyIioBackend()
+{
+	bool scan = false;
+	int backEndsCount = iio_get_backends_count();
+	for (int i = 0; i < backEndsCount; i++) {
+		QString backEnd(iio_get_backend(i));
+		if (backEnd.compare("xml")) {
+			QCheckBox *cb = new QCheckBox();
+			cb->setText(backEnd);
+			if (backEnd.compare("serial") == 0) {
+				connect(cb, &QCheckBox::toggled, this, [=](bool en) {
+					if (en) {
+						scanParamsList.push_back(backEnd + ":");
+					} else {
+						scanParamsList.removeOne(backEnd + ":");
+					}
+					ui->serialSettingsWidget->setEnabled(en);
```

Comment:
You could create the connect when you create the checkbox instead of dynamically connecting here.
Here you can just enable/disable the control